controller/handler_ctrl/bind_service_request.go

```diff
@@ -52,7 +52,7 @@ func (h *hostRequestHandler) HandleReceive(msg *channel2.Message, ch channel2.Ch
 
 	var err error
 	if request.BindType == ctrl_pb.BindType_Bind {
-		err = h.network.BindService(h.r, request.Token, request.ServiceId)
+		err = h.network.BindService(h.r, request.Token, request.ServiceId, request.Pubkey)
```

Comment:
Some places we have `Pubkey`, other places the `Key` is capitalized.

Would it be possible to just make the fabric-level API take an `interface{}`, and call it something more open-ended like `userData`? (I know you'd called it `hints` the first time).

Reply:
certainly we can do that, but I assume that this code would have to be refactored soon anyway for multi/HA-bindings (@plorenz ). 
for now we'd need to know to extract key anyway or implement a way to store any kind of data

Reply:
I'm sure it will be affected by those changes, but I'm also sure the concept will remain in the new implementation. So, we should still endeavor to find the right abstraction now.

Reply:
The HA work will add strategy options/inputs. At a first pass, what I'm thinking is that peerData could be map[string][]byte. The controller would extract the values it knows about and ignore anything else. So it wouldn't store arbitrary data.** Maybe on the controller side we deal with map[string][]byte, but in the SDK we provide specific APIs to allowing passing in pub key and strategyInput? That way the controller APIs are more static, and the SDK apis are more user friendly.

**Except strategyOptions are pretty arbitrary, have to figure out how to model that. Right now I'm thinking interpret the []byte as string JSON and decode it.